Microsoft.Toolkit.Uwp/Helpers/DispatcherQueueHelper.cs

```diff
@@ -0,0 +1,249 @@
+             * Also reuse Task.CompletedTask in case of success,
+             * to skip an unnecessary heap allocation for every invocation. */
+
+            // Ignoring for now, but need to map the CurrentThreadID for all dispatcher queue code we have
+            /*
+            if (dispatcher.HasThreadAccess)
+            {
+                try
+                {
+                    function();
+
+                    return Task.CompletedTask;
+                }
+                catch (Exception e)
+                {
+                    return Task.FromException(e);
+                }
+            }
+            */
+
+            var taskCompletionSource = new TaskCompletionSource<object>();
+
+            bool ok = dispatcher.TryEnqueue(priority, () =>
+            {
+                try
+                {
+                    function();
+
+                    taskCompletionSource.SetResult(null);
+                }
+                catch (Exception e)
+                {
+                    taskCompletionSource.SetException(e);
+                }
+            });
+
+            return taskCompletionSource.Task;
+        }
+
+        /// <summary>
+        /// Extension method for <see cref="DispatcherQueue"/>. Offering an actual awaitable <see cref="Task{T}"/> with optional result that will be executed on the given dispatcher.
+        /// </summary>
+        /// <typeparam name="T">Returned data type of the function.</typeparam>
+        /// <param name="dispatcher">DispatcherQueue of a thread to run <paramref name="function"/>.</param>
+        /// <param name="function"> Function to be executed on the given dispatcher.</param>
+        /// <param name="priority">DispatcherQueue execution priority, default is normal.</param>
+        /// <returns>An awaitable <see cref="Task{T}"/> for the operation.</returns>
+        /// <remarks>If the current thread has UI access, <paramref name="function"/> will be invoked directly.</remarks>
+        public static Task<T> ExecuteOnUIThreadAsync<T>(this DispatcherQueue dispatcher, Func<T> function, DispatcherQueuePriority priority = DispatcherQueuePriority.Normal)
+        {
+            if (function is null)
+            {
+                throw new ArgumentNullException(nameof(function));
+            }
+
+            // Skip the dispatch, if posssible
+            // Ignoring for now, but need to map the CurrentThreadID for all dispatcher queue code we have
+            /*
+            if (dispatcher.HasThreadAccess)
+            {
+                try
+                {
+                    return Task.FromResult(function());
+                }
+                catch (Exception e)
+                {
+                    return Task.FromException<T>(e);
+                }
+            }
+            */
+
+            var taskCompletionSource = new TaskCompletionSource<T>();
+
+            _ = dispatcher.TryEnqueue(priority, () =>
```

Comment:
>_ [](start = 12, length = 1)

As above - consider what you want to do if TryEnqueue returns false, you may not want to simply ignore the return value.

Reply:
Whatever we decide on one of these methods, they should have the same behavior. Let's use the topic above.